tests/data_context/test_data_context_utils.py

```diff
@@ -227,3 +230,20 @@ def test_password_masker_mask_db_url():
     # in-memory
     assert PasswordMasker.mask_db_url("sqlite://") == "sqlite://"
     assert PasswordMasker.mask_db_url("sqlite://", use_urlparse=True) == "sqlite://"
+
+
+def test_parse_substitution_variable():
+    """
+    What does this test and why?
+    Ensure parse_substitution_variable works as expected.
+    Returns:
+
+    """
+    assert parse_substitution_variable("${SOME_VAR}") == "SOME_VAR"
```

Comment:
These are awesome, too! I think a couple more test cases that would be nice to have would be 
```
assert parse_substitution_variable("${SOME_$TRING}") == "SOME_$TRING"
assert parse_substitution_variable("$SOME_$TRING") == "SOME_$TRING"
```

I believe that's how these would work, but I'm not certain, and test cases would clarify.

Reply:
Oh good catch. Actually as currently written these tests would produce the following output since having a $ in a substitution variable is not supported. I don't think we explicitly state it but substitution variables follow python naming conventions (only alphanumeric + "_", and can't start with a number). So the first one is not a valid substitution string so we return None, the second is two valid substitution strings and currently `parse_substitution_variable` just returns the first. The consequences of this I think are OK - if someone tries to save a multi-sub value string e.g. ${SOME_STRING}/${ANOTHER_STRING} then the first valid substitution string will trigger skipping escaping in `escape_all_config_variables` as long as `skip_if_substitution_variable` is True (default).
```
assert parse_substitution_variable("${SOME_$TRING}") is None
assert parse_substitution_variable("$SOME_$TRING") == "SOME_"
```

Reply:
Also, your comment led me to remove the "-" I was allowing in `parse_substitution_variable` to be consistent with python variable naming conventions, so thank you!

Reply:
I'm going to merge after tests pass, but if you think the above functionality (returning only the first substitution string for `parse_substitution_variable`) should be changed let me know and I can create a separate PR if this is already merged.

Reply:
Hi @anthonyburdi - that makes sense! Thanks for the clarification.

For the case of "$SOME_$TRING", I imagine that an outcome of "SOME_" would be unexpected (I can't see a user specifying this as "$SOME_$TRING" with the goal of producing "SOME_"). I wonder if it makes sense to raise an error here, or at least to log a warning? This seems like a rare edge case, so probably not a huge deal either way.